src/qinfer/hyper_heuristic_optimisers.py

```diff
@@ -0,0 +1,202 @@
+
+
+## FEATURES ###################################################################
+
+from __future__ import absolute_import
+from __future__ import division
+
+## IMPORTS ####################################################################
+
+import numpy as np
+import random
+from functools import partial
+from qinfer.perf_testing import perf_test_multiple
+from qinfer import distributions
+    
+## CLASSES ####################################################################
+
+__all__ = [
+    'ParticleSwarmOptimiser'
+]
+
+class HyperHeuristicOptimiser(object):
```

Comment:
At the level of generality that this class is written, including ``HyperHeuristic`` in the name may be slightly misleading? I think it makes a lot of sense to keep the default choice of function being the performance objective below, but to rename here as a nod to that generality. Perhaps even just ``Optimiser`` (or even ``Optimizer``, since the project was started in American English instead of something a bit more common)...